src/parser.rs

```diff
@@ -709,26 +709,26 @@ impl<'a> Parser<'a> {
         }
         let mut it = date.chars();
         let mut valid = true;
-        valid = valid && it.next().map(is_digit).unwrap_or(false);
```

Comment:
I tend to prefer `map(...).unwrap_or(...)` as the ordering of the arguments makes more sense to me, could these (and the above) changes be reverted?